modules/web/web-api/src/main/java/com/enonic/xp/web/vhost/ResolveVirtualHostResult.java

```diff
@@ -0,0 +1,152 @@
+package com.enonic.xp.web.vhost;
+
+import java.util.Objects;
+
+import com.enonic.xp.annotation.PublicApi;
+import com.enonic.xp.security.IdProviderKey;
+import com.enonic.xp.security.IdProviderKeys;
+
+@PublicApi
+public class ResolveVirtualHostResult
+    implements VirtualHost
```

Comment:
use composition instead. provide a field resolvedVirtualHost (even if it is a single field now)